backend/src/main/java/org/cryptomator/hub/api/cipherduck/storage/S3StorageHelper.java

```diff
@@ -0,0 +1,201 @@
+package org.cryptomator.hub.api.cipherduck.storage;
+
+
+import jakarta.ws.rs.ClientErrorException;
+import jakarta.ws.rs.core.Response;
+import org.cryptomator.hub.api.cipherduck.CreateS3STSBucketDto;
+import org.cryptomator.hub.api.cipherduck.StorageProfileS3STSDto;
+import org.jboss.logging.Logger;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3ClientBuilder;
+import software.amazon.awssdk.services.s3.S3Configuration;
+import software.amazon.awssdk.services.s3.model.AccelerateConfiguration;
+import software.amazon.awssdk.services.s3.model.BucketAccelerateStatus;
+import software.amazon.awssdk.services.s3.model.BucketVersioningStatus;
+import software.amazon.awssdk.services.s3.model.CreateBucketConfiguration;
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest;
+import software.amazon.awssdk.services.s3.model.GetBucketAccelerateConfigurationRequest;
+import software.amazon.awssdk.services.s3.model.GetBucketAccelerateConfigurationResponse;
+import software.amazon.awssdk.services.s3.model.GetBucketEncryptionRequest;
+import software.amazon.awssdk.services.s3.model.GetBucketEncryptionResponse;
+import software.amazon.awssdk.services.s3.model.GetBucketVersioningRequest;
+import software.amazon.awssdk.services.s3.model.GetBucketVersioningResponse;
+import software.amazon.awssdk.services.s3.model.HeadBucketRequest;
+import software.amazon.awssdk.services.s3.model.NoSuchBucketException;
+import software.amazon.awssdk.services.s3.model.PutBucketAccelerateConfigurationRequest;
+import software.amazon.awssdk.services.s3.model.PutBucketEncryptionRequest;
+import software.amazon.awssdk.services.s3.model.PutBucketVersioningRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.services.s3.model.ServerSideEncryption;
+import software.amazon.awssdk.services.s3.model.ServerSideEncryptionByDefault;
+import software.amazon.awssdk.services.s3.model.ServerSideEncryptionConfiguration;
+import software.amazon.awssdk.services.s3.model.ServerSideEncryptionRule;
+import software.amazon.awssdk.services.s3.model.VersioningConfiguration;
+
+import java.net.URI;
+import java.util.Base64;
+import java.util.Collections;
+
+public class S3StorageHelper {
+	private static final Logger log = Logger.getLogger(S3StorageHelper.class);
+
+	public static void makeS3Bucket(
+			final StorageProfileS3STSDto storageConfig,
+			final CreateS3STSBucketDto dto
+	) {
+
+		if (log.isInfoEnabled()) {
+			log.info(String.format("Make S3 bucket %s for profile %s", dto, storageConfig));
+		}
+
+		final String bucketName = storageConfig.bucketPrefix() + dto.vaultId();
```

Comment:
Why not have the actual bucket name in the storage configuration instead of a prefix only?